src/v/cloud_storage/partition_manifest.cc

```diff
@@ -503,6 +510,85 @@ bool partition_manifest::contains(const segment_name& name) const {
 
 void partition_manifest::delete_replaced_segments() { _replaced.clear(); }
 
+model::offset partition_manifest::get_archive_start_offset() const {
+    return _archive_start_offset;
+}
+
+model::offset_delta partition_manifest::get_archive_start_offset_delta() const {
+    return _archive_start_offset_delta;
+}
+
+kafka::offset partition_manifest::get_archive_start_kafka_offset() const {
+    if (_archive_start_offset == model::offset{}) {
+        return kafka::offset{};
+    }
+    return _archive_start_offset - _archive_start_offset_delta;
+}
+
+model::offset partition_manifest::get_archive_clean_offset() const {
+    return _archive_clean_offset;
+}
+
+void partition_manifest::set_archive_start_offset(
+  model::offset start_rp_offset, model::offset_delta start_delta) {
+    if (_archive_start_offset < start_rp_offset) {
```

Comment:
nit: an error/warn log for when this branch is **not** taken would be nice

Reply:
done